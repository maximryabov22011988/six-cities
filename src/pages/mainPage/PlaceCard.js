import React from 'react';
import PropTypes from 'prop-types';

import Rating from './placeCard/Rating';

const propTypes = PropTypes.shape({
  card: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    image: PropTypes.string,
    price: PropTypes.number,
    type: PropTypes.oneOf(['Apartment', 'Private room']),
    rating: PropTypes.number,
    isPremium: PropTypes.bool,
    isBookmark: PropTypes.bool
  }),
  onTitleClick: PropTypes.func
}).isRequired;

function PlaceCard(props) {
  const { card, onTitleClick } = props;
  const { title, image, price, type, rating, isPremium, isBookmark } = card;

  return (
    <article className="cities__place-card place-card">
      {isPremium && (
        <div className="place-card__mark">
          <span>Premium</span>
        </div>
      )}

      <div className="cities__image-wrapper place-card__image-wrapper">
        <a href="#">
          <img
            className="place-card__image"
            src={image}
            width="260"
            height="200"
            alt="Place image"
          />
        </a>
      </div>

      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>
          <button
            className={`${isBookmark &&
              'place-card__bookmark-button--active'} place-card__bookmark-button button`}
            type="button"
          >
            <svg className="place-card__bookmark-icon" width="18" height="19">
              <use xlinkHref="#icon-bookmark" />
            </svg>
            <span className="visually-hidden">To bookmarks</span>
          </button>
        </div>

        <Rating rating={rating} />

        <h2 className="place-card__name">
          <a href="#" onClick={onTitleClick}>
            {title}
          </a>
        </h2>

        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
}

PlaceCard.propTypes = propTypes;

export default PlaceCard;
