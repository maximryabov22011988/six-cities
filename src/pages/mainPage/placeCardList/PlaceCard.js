import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Rating from './placeCard/Rating';

const propTypes = PropTypes.shape({
  offer: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    image: PropTypes.string,
    price: PropTypes.number,
    type: PropTypes.oneOf(['Apartment', 'Private room']),
    rating: PropTypes.number,
    isPremium: PropTypes.bool,
    isBookmark: PropTypes.bool
  }),
  onTitleClick: PropTypes.func,
  onImageClick: PropTypes.func,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func
}).isRequired;

function PlaceCard(props) {
  const {
    className,
    offer,
    onTitleClick,
    onImageClick,
    onMouseEnter,
    onMouseLeave
  } = props;
  const { title, image, price, type, rating, isPremium, isBookmark } = offer;

  return (
    <article
      className={classnames('cities__place-card place-card', className)}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {isPremium && (
        <div className="place-card__mark">
          <span>Premium</span>
        </div>
      )}

      <div className="cities__image-wrapper place-card__image-wrapper">
        <a href="#" onClick={onImageClick(offer)}>
          <img
            className="place-card__image"
            src={image}
            width="260"
            height="200"
            alt="Place image"
          />
        </a>
      </div>

      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>
          <button
            className={classnames(
              'place-card__bookmark-button button',
              isBookmark && 'place-card__bookmark-button--active'
            )}
            type="button"
          >
            <svg className="place-card__bookmark-icon" width="18" height="19">
              <use xlinkHref="#icon-bookmark" />
            </svg>
            <span className="visually-hidden">To bookmarks</span>
          </button>
        </div>

        <Rating rating={rating} />

        <h2 className="place-card__name">
          <a href="#" onClick={onTitleClick}>
            {title}
          </a>
        </h2>

        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
}

PlaceCard.propTypes = propTypes;

export default PlaceCard;
