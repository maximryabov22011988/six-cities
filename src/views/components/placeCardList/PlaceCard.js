import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cn from 'classnames';

import Label from '../Label';
import Button from '../Button';
import Image from '../Image';
import Rating from '../Rating';
import SvgIcon from '../SvgIcon';

import withActiveItem from '../../hocs/withActiveItem';

const propTypes = {
  className: PropTypes.string,
  offer: PropTypes.shape({
    bedrooms: PropTypes.number,
    city: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.object,
    }),
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatar_url: PropTypes.string,
      id: PropTypes.number,
      is_pro: PropTypes.bool,
      name: PropTypes.string,
    }),
    id: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    is_favorite: PropTypes.bool,
    is_premium: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
    max_adults: PropTypes.number,
    preview_image: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string,
  }).isRequired,
  onActiveOfferClick: PropTypes.func.isRequired,
};

function PlaceCard({ className, offer, onActiveOfferClick, onMouseEnter, onMouseLeave }) {
  const {
    id,
    title,
    preview_image: previewImage,
    price,
    type,
    rating,
    is_premium: isPremium,
    is_favorite: isFavorite,
  } = offer;

  const BookmarkButton = withActiveItem(Button);

  return (
    <article
      className={cn(
        'place-card',
        className === 'cities' && `${className}__place-card`,
        className === 'near-places' && `${className}__card`,
      )}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      <Label isShow={isPremium} name="Premium" parentClassName="place-card" />

      <div className={`${className}__image-wrapper place-card__image-wrapper`}>
        <a href="#" onClick={onActiveOfferClick}>
          <Image className="place-card__image" height="200" label="Place image" src={previewImage} width="260" />
        </a>
      </div>

      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>

          <BookmarkButton className="place-card__bookmark-button" isActive={isFavorite}>
            <SvgIcon
              className="place-card__bookmark-icon"
              height="19"
              isShowLabel={false}
              label={isFavorite ? 'In bookmarks' : 'To bookmarks'}
              name="icon-bookmark"
              width="18"
            />
          </BookmarkButton>
        </div>

        <Rating parentClassName="place-card" rating={rating} />

        <h2 className="place-card__name">
          <Link to={`/offer/${id}`}>{title}</Link>
        </h2>

        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
}

PlaceCard.propTypes = propTypes;

export default PlaceCard;
