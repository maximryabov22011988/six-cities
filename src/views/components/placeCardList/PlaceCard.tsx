import * as React from 'react';
import { Link } from 'react-router-dom';
import cn from 'classnames';

import Label from '../Label';
import Button from '../Button';
import Image from '../Image';
import Rating from '../Rating';
import SvgIcon from '../SvgIcon';

import withActiveItem from '../../hocs/withActiveItem';

import { Offer, onAddToFavorities, onRemoveFromFavorities } from '../../interfaces';

interface Props {
  className: string,
  offer: Offer,
  onActiveOfferClick: ((event: React.MouseEvent<HTMLAnchorElement>) => void) | undefined,
  onAddToFavorities: onAddToFavorities,
  onRemoveFromFavorities: onRemoveFromFavorities,
}

function PlaceCard({
  className,
  offer,
  onActiveOfferClick,
  onAddToFavorities,
  onRemoveFromFavorities,
}: Props) {
  const {
    id,
    title,
    preview_image: previewImage,
    price,
    type,
    rating,
    is_premium: isPremium,
    is_favorite: isFavorite,
  } = offer;

  const BookmarkButton = withActiveItem(Button);

  return (
    <article
      className={cn(
        'place-card',
        className === 'cities' && `${className}__place-card`,
        className === 'near-places' && `${className}__card`,
        className === 'favorites' && `${className}__card`
      )}
    >
      <Label isShow={isPremium} name="Premium" parentClassName="place-card" />

      <div className={`${className}__image-wrapper place-card__image-wrapper`}>
        <a href="#" onClick={onActiveOfferClick}>
          <Image
            className="place-card__image"
            height={className === 'favorites' ? '110' : '200'}
            label="Place image"
            src={previewImage}
            width={className === 'favorites' ? '150' : '260'}
          />
        </a>
      </div>

      <div className={cn('place-card__info', className === 'favorites' && `${className}__card-info`)}>
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>

          <BookmarkButton
            className="place-card__bookmark-button"
            isActive={isFavorite}
            onClick={isFavorite ? onRemoveFromFavorities(id) : onAddToFavorities(id)}
          >
            <SvgIcon
              className="place-card__bookmark-icon"
              height="19"
              isShowLabel={false}
              label={isFavorite ? 'In bookmarks' : 'To bookmarks'}
              name="icon-bookmark"
              width="18"
            />
          </BookmarkButton>
        </div>

        <Rating parentClassName="place-card" rating={rating} />

        <h2 className="place-card__name">
          <Link to={`/offer/${id}`}>{title}</Link>
        </h2>

        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
}

export default PlaceCard;
