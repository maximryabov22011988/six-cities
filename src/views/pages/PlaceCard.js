import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import Header from '../components/Header';
import Logo from '../components/Logo';
import UserInfo from '../components/UserInfo';
import Label from '../components/Label';
import Rating from '../components/Rating';
import Button from '../components/Button';
import SvgIcon from '../components/SvgIcon';
import Avatar from '../components/Avatar';
import Map from '../components/Map';

import Reviews from './placeCard/Reviews';
import NearPlaces from './placeCard/NearPlaces';
import AddReviewsForm from './placeCard/AddReviewsForm';

import withActiveItem from '../hocs/withActiveItem';

import { sendReview } from '../../state/reviews/operations';
import { getCurrentOffer, getNearOffers } from '../../state/offers/selectors';

const MAX_IMAGES = 6;

const propTypes = {
  addOfferToFavorities: PropTypes.func.isRequired,
  isAuthUser: PropTypes.bool.isRequired,
  nearOffers: PropTypes.arrayOf(PropTypes.object),
  offer: PropTypes.shape({
    bedrooms: PropTypes.number,
    city: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.object,
    }),
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatar_url: PropTypes.string,
      id: PropTypes.number,
      is_pro: PropTypes.bool,
      name: PropTypes.string,
    }),
    id: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    is_favorite: PropTypes.bool,
    is_premium: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
    max_adults: PropTypes.number,
    preview_image: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string,
  }).isRequired,
  removeOfferFromFavorities: PropTypes.func.isRequired,
  sendReview: PropTypes.func.isRequired,
  user: PropTypes.shape({
    id: PropTypes.number,
    email: PropTypes.string,
    name: PropTypes.string,
    avatar_url: PropTypes.string,
    is_pro: PropTypes.bool,
  }).isRequired,
};

function PlaceCard({
  isAuthUser,
  user: { avatar_url: userAvatarUrl, email },
  nearOffers,
  offer,
  sendReview,
  addOfferToFavorities,
  removeOfferFromFavorities,
}) {
  const {
    id,
    images,
    title,
    price,
    type,
    bedrooms,
    host: { avatar_url: avatarUrl, is_pro: isPro, name },
    is_favorite: isFavorite,
    is_premium: isPremium,
    max_adults: maxAdults,
    rating,
    goods,
    description,
  } = offer;

  const BookmarkButton = withActiveItem(Button);

  const handleAddToFavorities = (hotelId) => () => addOfferToFavorities(hotelId);
  const handleRemoveFromFavorities = (hotelId) => () => removeOfferFromFavorities(hotelId);

  return (
    <div className="page">
      <Header
        logo={<Logo position="header" />}
        userInfo={<UserInfo avatarUrl={userAvatarUrl} email={email} isAuth={isAuthUser} />}
      />

      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {images.map((src, i) => {
                if (i < MAX_IMAGES) {
                  return (
                    <div className="property__image-wrapper" key={i}>
                      <img alt="Hotel interior" className="property__image" src={src} />
                    </div>
                  );
                }
                return null;
              })}
            </div>
          </div>

          <div className="property__container container">
            <div className="property__wrapper">
              <Label isShow={isPremium} name="Premium" parentClassName="property" />

              <div className="property__name-wrapper">
                <h1 className="property__name">{title}</h1>
                <BookmarkButton
                  className="property__bookmark-button"
                  isActive={isFavorite}
                  onClick={isFavorite ? handleRemoveFromFavorities(id) : handleAddToFavorities(id)}
                >
                  <SvgIcon
                    className="property__bookmark-icon"
                    height="33"
                    isShowLabel={false}
                    label={isFavorite ? 'In bookmarks' : 'To bookmarks'}
                    name="icon-bookmark"
                    width="31"
                  />
                </BookmarkButton>
              </div>

              <Rating parentClassName="property" rating={rating} showValue />

              <ul className="property__features">
                <li className="property__feature property__feature--entire">{type}</li>
                <li className="property__feature property__feature--bedrooms">{bedrooms} Bedrooms</li>
                <li className="property__feature property__feature--adults">Max {maxAdults} adults</li>
              </ul>

              <div className="property__price">
                <b className="property__price-value">&euro;{price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>

              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {goods.map((good, i) => (
                    <li className="property__good-item" key={i}>
                      {good}
                    </li>
                  ))}
                </ul>
              </div>

              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>

                <div className="property__host-user user">
                  <Avatar
                    alt="Host avatar"
                    height="74"
                    isPro={isPro}
                    name={name}
                    parentClassName="property"
                    showStatus
                    src={avatarUrl}
                    width="74"
                  />
                </div>

                <div className="property__description">
                  <p className="property__text">{description}</p>
                </div>
              </div>

              <section className="property__reviews reviews">
                <Reviews hotelId={id} />
                {isAuthUser && <AddReviewsForm hotelId={id} onSendReview={sendReview} />}
              </section>
            </div>
          </div>
          <Map className="property__map" currentOffer={offer} fixed offers={nearOffers} />
        </section>

        <NearPlaces
          offers={nearOffers}
          onAddToFavorities={handleAddToFavorities}
          onRemoveFromFavorities={handleRemoveFromFavorities}
        />
      </main>
    </div>
  );
}

PlaceCard.propTypes = propTypes;

const mapStateToProps = (state, ownProps) => ({
  offer: getCurrentOffer(state, ownProps),
  nearOffers: getNearOffers(state, ownProps),
});

const mapDispatchToProps = {
  sendReview,
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(PlaceCard)
);
