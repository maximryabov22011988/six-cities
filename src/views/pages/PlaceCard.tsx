import * as React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import Header from '../components/Header';
import Logo from '../components/Logo';
import UserInfo from '../components/UserInfo';
import Label from '../components/Label';
import Rating from '../components/Rating';
import Button from '../components/Button';
import SvgIcon from '../components/SvgIcon';
import Avatar from '../components/Avatar';
import Map from '../components/Map';

import Reviews from './placeCard/Reviews';
import NearPlaces from './placeCard/NearPlaces';
import AddReviewsForm from './placeCard/AddReviewsForm';

import withActiveItem from '../hocs/withActiveItem';

import { sendReview } from '../../state/reviews/operations';
import { getCurrentOffer, getNearOffers } from '../../state/offers/selectors';

import { User, Offer, onAddToFavorities, onRemoveFromFavorities } from '../interfaces';

const MAX_IMAGES = 6;

interface Props {
  addOfferToFavorities: onAddToFavorities,
  isAuthUser: boolean,
  nearOffers: Array<Offer>,
  offer: Offer,
  removeOfferFromFavorities: onRemoveFromFavorities,
  sendReview: () => void,
  user:User,
}

function PlaceCard({
  isAuthUser,
  user: { avatar_url: userAvatarUrl, email },
  nearOffers,
  offer,
  sendReview,
  addOfferToFavorities,
  removeOfferFromFavorities,
}: Props) {
  const {
    id,
    images,
    title,
    price,
    type,
    bedrooms,
    host: { avatar_url: avatarUrl, is_pro: isPro, name },
    is_favorite: isFavorite,
    is_premium: isPremium,
    max_adults: maxAdults,
    rating,
    goods,
    description,
  } = offer;

  const BookmarkButton = withActiveItem(Button);

  const handleAddToFavorities = (hotelId) => () => addOfferToFavorities(hotelId);
  const handleRemoveFromFavorities = (hotelId) => () => removeOfferFromFavorities(hotelId);

  return (
    <div className="page">
      <Header
        logo={<Logo position="header" />}
        userInfo={<UserInfo avatarUrl={userAvatarUrl} email={email} isAuth={isAuthUser} />}
      />

      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {images.map((src, i) => {
                if (i < MAX_IMAGES) {
                  return (
                    <div className="property__image-wrapper" key={i}>
                      <img alt="Hotel interior" className="property__image" src={src} />
                    </div>
                  );
                }
                return null;
              })}
            </div>
          </div>

          <div className="property__container container">
            <div className="property__wrapper">
              <Label isShow={isPremium} name="Premium" parentClassName="property" />

              <div className="property__name-wrapper">
                <h1 className="property__name">{title}</h1>
                <BookmarkButton
                  className="property__bookmark-button"
                  isActive={isFavorite}
                  onClick={isFavorite ? handleRemoveFromFavorities(id) : handleAddToFavorities(id)}
                >
                  <SvgIcon
                    className="property__bookmark-icon"
                    height="33"
                    isShowLabel={false}
                    label={isFavorite ? 'In bookmarks' : 'To bookmarks'}
                    name="icon-bookmark"
                    width="31"
                  />
                </BookmarkButton>
              </div>

              <Rating isShowValue parentClassName="property" rating={rating} />

              <ul className="property__features">
                <li className="property__feature property__feature--entire">{type}</li>
                <li className="property__feature property__feature--bedrooms">{bedrooms} Bedrooms</li>
                <li className="property__feature property__feature--adults">Max {maxAdults} adults</li>
              </ul>

              <div className="property__price">
                <b className="property__price-value">&euro;{price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>

              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {goods.map((good, i) => (
                    <li className="property__good-item" key={i}>
                      {good}
                    </li>
                  ))}
                </ul>
              </div>

              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>

                <div className="property__host-user user">
                  <Avatar
                    alt="Host avatar"
                    height="74"
                    isPro={isPro}
                    name={name}
                    parentClassName="property"
                    showStatus
                    src={avatarUrl}
                    width="74"
                  />
                </div>

                <div className="property__description">
                  <p className="property__text">{description}</p>
                </div>
              </div>

              <section className="property__reviews reviews">
                <Reviews hotelId={id} />
                {isAuthUser && <AddReviewsForm hotelId={id} onSendReview={sendReview} />}
              </section>
            </div>
          </div>
          <Map className="property__map" currentOffer={offer} fixed offers={nearOffers} />
        </section>

        <NearPlaces
          offers={nearOffers}
          onAddToFavorities={handleAddToFavorities}
          onRemoveFromFavorities={handleRemoveFromFavorities}
        />
      </main>
    </div>
  );
}

const mapStateToProps = (state, ownProps) => ({
  offer: getCurrentOffer(state, ownProps),
  nearOffers: getNearOffers(state, ownProps),
});

const mapDispatchToProps = {
  sendReview,
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(PlaceCard)
);
