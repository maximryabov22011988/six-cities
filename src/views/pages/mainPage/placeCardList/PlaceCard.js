import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import withActiveItem from '../../../hocs/withActiveItem';
import Button from '../../../components/Button';
import Rating from './placeCard/Rating';

const propTypes = {
  offer: PropTypes.shape({
    bedrooms: PropTypes.number,
    city: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.object
    }),
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatar_url: PropTypes.string,
      id: PropTypes.number,
      is_pro: PropTypes.bool,
      name: PropTypes.string
    }),
    id: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    is_favorite: PropTypes.bool,
    is_premium: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number
    }),
    max_adults: PropTypes.number,
    preview_image: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string
  }).isRequired,
  onTitleClick: PropTypes.func.isRequired,
  onImageClick: PropTypes.func.isRequired,
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired
};

function PlaceCard(props) {
  const {
    className,
    offer,
    onTitleClick,
    onImageClick,
    onMouseEnter,
    onMouseLeave
  } = props;

  const {
    title,
    preview_image: previewImage,
    price,
    type,
    rating,
    is_premium: isPremium,
    is_favorite: isFavorite
  } = offer;

  const BookmarkButton = withActiveItem(Button);

  return (
    <article
      className={classnames('cities__place-card place-card', className)}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {isPremium && (
        <div className="place-card__mark">
          <span>Premium</span>
        </div>
      )}

      <div className="cities__image-wrapper place-card__image-wrapper">
        <a href="#" onClick={onImageClick(offer)}>
          <img
            className="place-card__image"
            src={previewImage}
            width="260"
            height="200"
            alt="Place image"
          />
        </a>
      </div>

      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>

          <BookmarkButton
            className="place-card__bookmark-button"
            isActive={isFavorite}
          >
            <svg className="place-card__bookmark-icon" width="18" height="19">
              <use xlinkHref="#icon-bookmark" />
            </svg>
            <span className="visually-hidden">To bookmarks</span>
          </BookmarkButton>
        </div>

        <Rating rating={rating} />

        <h2 className="place-card__name">
          <a href="#" onClick={onTitleClick}>
            {title}
          </a>
        </h2>

        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
}

PlaceCard.propTypes = propTypes;

export default PlaceCard;
