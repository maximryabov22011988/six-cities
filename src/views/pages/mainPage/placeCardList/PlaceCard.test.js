import React from 'react';
import renderer from 'react-test-renderer';

import PlaceCard from './PlaceCard';

const offerMock = {
  id: 1,
  city: {
    name: 'Cologne',
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  title: 'The Pondhouse - A Magical Place',
  preview_image: `img/apartment-01.jpg`,
  price: 120,
  type: 'house',
  rating: 4.6,
  is_premium: true,
  is_favorite: false
};

it('PlaceCard correctly render', () => {
  const handleTitleClick = jest.fn();
  const handleImageClick = jest.fn();
  const handleMouseEnter = jest.fn();
  const handleMouseLeave = jest.fn();

  const placeCard = renderer
    .create(
      <PlaceCard
        offer={offerMock}
        onTitleClick={handleTitleClick}
        onImageClick={handleImageClick}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
      />
    )
    .toJSON();

  expect(placeCard).toMatchSnapshot();
});
