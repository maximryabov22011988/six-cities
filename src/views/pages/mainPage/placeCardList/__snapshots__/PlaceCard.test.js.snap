// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard correctly render 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[MockFunction]}
  onMouseLeave={[MockFunction]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "background": "#e1e1e1",
            "borderRadius": "4px",
            "display": "flex",
            "height": "200px",
            "justifyContent": "center",
            "width": "260px",
          }
        }
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={0}
          onError={[Function]}
          onLoad={[Function]}
          src="img/apartment-01.jpg"
          width={0}
        />
        <span>
          Loading ...
        </span>
      </div>
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          120
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "92.00%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[MockFunction]}
      >
        The Pondhouse - A Magical Place
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      house
    </p>
  </div>
</article>
`;
